core/src/main/scala/org/broadinstitute/dsde/rawls/fastpass/FastPassService.scala

```diff
@@ -0,0 +1,444 @@
+
+import cats.effect.IO
+import cats.effect.unsafe.implicits.global
+import com.typesafe.scalalogging.LazyLogging
+import org.broadinstitute.dsde.rawls.config.FastPassConfig
+import org.broadinstitute.dsde.rawls.dataaccess.SamDAO
+import org.broadinstitute.dsde.rawls.model.{
+  FastPassGrant,
+  GoogleProjectId,
+  RawlsRequestContext,
+  RawlsUserEmail,
+  RawlsUserSubjectId,
+  SamResourcePolicyName,
+  SamResourceRole,
+  SamResourceTypeNames,
+  SamWorkspaceRoles,
+  Workspace
+}
+import org.broadinstitute.dsde.workbench.google.{GoogleIamDAO, GoogleStorageDAO}
+import org.broadinstitute.dsde.workbench.model.WorkbenchEmail
+import org.broadinstitute.dsde.workbench.model.google.{GcsBucketName, GoogleProject}
+import org.broadinstitute.dsde.rawls.dataaccess.slick.{DataAccess, ReadWriteAction}
+import org.broadinstitute.dsde.rawls.fastpass.FastPassService.{
+  maxBucketRoleBindingsPerUser,
+  maxProjectRoleBindingsPerUser,
+  policyBindingsQuotaLimit
+}
+import org.broadinstitute.dsde.rawls.util.TracingUtils.traceDBIOWithParent
+import org.joda.time.{DateTime, DateTimeZone}
+import slick.dbio.{DBIO, DBIOAction, Effect, NoStream}
+import org.broadinstitute.dsde.workbench.google.HttpGoogleIamDAO.fromProjectPolicy
+import org.broadinstitute.dsde.workbench.google.HttpGoogleStorageDAO.fromBucketPolicy
+import org.broadinstitute.dsde.workbench.model.google.iam.IamMemberTypes.IamMemberType
+import org.broadinstitute.dsde.workbench.model.google.iam.IamResourceTypes.IamResourceType
+import org.broadinstitute.dsde.workbench.model.google.iam.{Expr, IamMemberTypes, IamResourceTypes}
+import org.broadinstitute.dsde.workbench.openTelemetry.OpenTelemetryMetrics
+
+import scala.concurrent.{ExecutionContext, Future}
+
+object FastPassService {
+  def constructor(config: FastPassConfig,
+                  googleIamDao: GoogleIamDAO,
+                  googleStorageDAO: GoogleStorageDAO,
+                  samDAO: SamDAO,
+                  terraBillingProjectOwnerRole: String,
+                  terraWorkspaceCanComputeRole: String,
+                  terraWorkspaceNextflowRole: String,
+                  terraBucketReaderRole: String,
+                  terraBucketWriterRole: String
+  )(ctx: RawlsRequestContext, dataAccess: DataAccess)(implicit
+    executionContext: ExecutionContext,
+    openTelemetry: OpenTelemetryMetrics[IO]
+  ): FastPassService =
+    new FastPassService(
+      ctx,
+      dataAccess,
+      config,
+      googleIamDao,
+      googleStorageDAO,
+      samDAO,
+      terraBillingProjectOwnerRole,
+      terraWorkspaceCanComputeRole,
+      terraWorkspaceNextflowRole,
+      terraBucketReaderRole,
+      terraBucketWriterRole
+    )
+
+  val policyBindingsQuotaLimit = 1500
+  val maxProjectRoleBindingsPerUser = 6 // 3 possible roles for each user and pet service account
+  val maxBucketRoleBindingsPerUser = 2 // 1 possible role for each user and pet service account
+
+}
+
+class FastPassService(protected val ctx: RawlsRequestContext,
```

Comment:
add some doc about what FastPass is. It is a nice name but is not immediately apparent what it means to someone not in the know.